modules/sdk-coin-ton/src/tonToken.ts

```diff
@@ -0,0 +1,105 @@
+import { BitGoBase, CoinConstructor, NamedCoinConstructor, VerifyTransactionOptions } from '@bitgo/sdk-core';
+import BigNumber from 'bignumber.js';
+import { coins, TonTokenConfig, NetworkType, tokens } from '@bitgo/statics';
+
+import { Transaction } from './lib';
+import { Ton } from './ton';
+
+export class TonToken extends Ton {
+  public readonly tokenConfig: TonTokenConfig;
+
+  constructor(bitgo: BitGoBase, tokenConfig: TonTokenConfig) {
+    const staticsCoin = tokenConfig.network === NetworkType.MAINNET ? coins.get('ton') : coins.get('tton');
+    super(bitgo, staticsCoin);
+    this.tokenConfig = tokenConfig;
+  }
+
+  static createTokenConstructor(config: TonTokenConfig): CoinConstructor {
+    return (bitgo: BitGoBase) => new TonToken(bitgo, config);
+  }
+
+  static createTokenConstructors(
+    tokenConfig: TonTokenConfig[] = [...tokens.bitcoin.ton.tokens, ...tokens.testnet.ton.tokens]
+  ): NamedCoinConstructor[] {
+    const tokensCtors: NamedCoinConstructor[] = [];
+    for (const token of tokenConfig) {
+      const tokenConstructor = TonToken.createTokenConstructor(token);
+      tokensCtors.push({ name: token.type, coinConstructor: tokenConstructor });
+    }
+    return tokensCtors;
+  }
+
+  get name(): string {
+    return this.tokenConfig.name;
+  }
+
+  get coin(): string {
+    return this.tokenConfig.coin;
+  }
+
+  get jettonMaster(): string {
+    return this.tokenConfig.jettonMaster;
+  }
+
+  get decimalPlaces(): number {
+    return this.tokenConfig.decimalPlaces;
+  }
+
+  getChain(): string {
+    return this.tokenConfig.type;
+  }
+
+  getBaseChain(): string {
+    return this.coin;
+  }
+
+  getFullName(): string {
+    return 'TON Token';
+  }
+
+  getBaseFactor(): number {
+    return Math.pow(10, this.tokenConfig.decimalPlaces);
+  }
+
+  async verifyTransaction(params: VerifyTransactionOptions): Promise<boolean> {
```

Comment:
how will the verification work for different sub-transactions of a jetton transaction?